svc/FetchService.js

```diff
@@ -232,12 +232,28 @@ export class FetchService extends HoistService {
 
         if (!ret.ok) {
             ret.responseText = await this.safeResponseTextAsync(ret);
-            throw Exception.fetchError(opts, ret);
+            const e = Exception.fetchError(opts, ret);
+            if (!XH.isSSO && isRelativeUrl && e.httpStatus === 401) {
+                await this.maybeReloadForAuthAsync();
+            }
+            throw e;
         }
 
         return ret;
     }
 
+    async maybeReloadForAuthAsync() {
+        // Reload app to renew auth. Be careful not to trigger during
+        // initialization, or to get caught in a tight reload loop
+        const {appState, localStorageService} = XH;
+        if (appState !== AppState.RUNNING && appState !== AppState.SUSPENDED) return;
+        if (!olderThan(localStorageService.get('xhLastAuthReload', null), ONE_MINUTE)) return;
+
+        localStorageService.set('xhLastAuthReload', Date.now());
+        XH.reloadApp();
+        await never();
+    }
+
```

Comment:
What about showing the user a brief message to indicate what's happening - no action required, and might be a bit less mysterious. 

I think this pattern of reloading to prompt re-auth is relatively common, so it shouldn't be a shock in the intended use case where the user is returned to a login prompt, but in case anything doesn't work quite as planned this could also help us understand what a user is reporting around unexpected reloads.

```suggestion
    // Reload app to renew auth.
    async maybeReloadForAuthAsync() {
        const {appState, localStorageService} = XH;
        // Be careful not to trigger during initialization...
        if (appState !== AppState.RUNNING && appState !== AppState.SUSPENDED) return;
        // ...and avoid any chance of a tight reload loop.
        if (!olderThan(localStorageService.get('xhLastAuthReload', null), ONE_MINUTE)) return;

        XH.alert({
            title: 'Authentication Expired',
            message: fragment(
                `We're sorry, but you must log on again to continue using ${XH.clientAppName}.`,
                br(),
                `This page will reload momentarily...`
            ),
            confirmProps: {text: 'Reloading...', disabled: true}
        });
        await wait(5000);

        localStorageService.set('xhLastAuthReload', Date.now());
        XH.reloadApp();
        await never();
    }
```